modules/frontend/auth/data/src/hooks/useAuthConfirmPasswordByToken.tsx

```diff
@@ -0,0 +1,88 @@
+import { Text, useToast } from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import React from 'react';
+import { useMutation, UseMutationOptions, useQueryClient } from 'react-query';
+
+import { QUERY_KEYS } from '../constants/querykes';
+import { ConfirmPasswordDto } from '../dto/confirmPasswordDto';
+import { DefaultResponseDto } from '../dto/DefaultResponseDto';
+import AuthService from '../services/authService';
+
+export const useAuthConfirmPasswordByToken = () => {
+  const queryClient = useQueryClient();
+  const authService = AuthService();
+  const toast = useToast();
+  const router = useRouter();
+
+  const mutationRequest = React.useCallback(
+    async function mutationRequestWrapper(requestData: ConfirmPasswordDto): Promise<DefaultResponseDto> {
+      const { token } = router.query;
+      if (token) {
+        return authService.confirmPasswordByToken(token.toString(), requestData);
```

Comment:
We really need this .tostring() it isn't already a string?
```suggestion
        return authService.confirmPasswordByToken(token, requestData);
```